app/controllers/internal/profile_fields_controller.rb

```diff
@@ -1,32 +1,50 @@
 module Internal
   class ProfileFieldsController < Internal::ApplicationController
+    ALLOWED_PARAMS = %i[
+      input_type label active placeholder_text description
+    ].freeze
     layout "internal"
 
     def index
       @profile_fields = ProfileField.all
     end
 
     def update
-      @profile_fields = ProfileField.find(params[:id])
-      @profile_fields.update!(profile_field_params)
+      profile_field = ProfileField.find(params[:id])
+      if profile_field.update(profile_field_params)
+        flash[:success] = "Profile field #{profile_field.label} updated"
+      else
+        flash[:error] = "Error: #{profile_field.errors_as_sentence}"
+      end
       redirect_to internal_profile_fields_path
     end
 
     def create
-      @profile_field = ProfileField.create(profile_field_params)
+      profile_field = ProfileField.new(profile_field_params)
+      if profile_field.save
+        flash[:success] = "Profile field #{profile_field.label} created"
+      else
+        flash[:error] = "Error: #{profile_field.errors_as_sentence}"
+      end
       redirect_to internal_profile_fields_path
     end
 
     def destroy
-      @profile_field = ProfileField.find(params[:id])
-      @profile_field.destroy
+      profile_field = ProfileField.find(params[:id])
+      if profile_field.destroy
+        flash[:success] = "Profile field #{profile_field.label} deleted"
+      else
+        flash[:error] = "Error: #{profile_field.errors_as_sentence}"
+      end
       redirect_to internal_profile_fields_path
     end
 
     private
 
+    private_constant :ALLOWED_PARAMS
+
     def profile_field_params
-      allowed_params = %i[input_type label active placeholder_text description]
+      allowed_params = ALLOWED_PARAMS
```

Comment:
I just used the `ALLOWED_PARAMS`-related code as inspiration for something I'm currently working on--this is exactly what I needed! 😄